packages/teleport/src/lib/term/terminal.js

```diff
@@ -54,6 +55,19 @@ class TtyTerminal {
       cursorBlink: false,
     });
 
+    // TODO deprecated, use attachCustomKeyEventHandler when we upgrade xterm
+    this.term.attachCustomKeydownHandler(event => {
```

Comment:
Actually, we can keep this code closer to the feature. We can move it [here](https://github.com/gravitational/webapps/blob/master/packages/teleport/src/console/components/DocumentSsh/Terminal/Terminal.tsx#L29) and subscribe to xterm events from there.

Then we can add a helper method to useKeyboardNav.ts and export it.

```js

import { getPlatform } from 'design/theme/utils';

export function getMappedAction(event) {  
  const index = ['1', '2', '3', '4', '5', '6', '7', '8', '9'].indexOf(
    event.key
  );

  const { isMac } = getPlatform();
  const isModifierKey = (isMac && event.ctrlKey) || event.altKey;
  if (!isModifierKey || index == -1) {
    return {};
  }

  return {
    tabSwitch: {
      index
    },
  };
}
```

for attachCustomKeydownHandler:

```js
const action = getMappedAction(event)
if (!action.tabSwitch){
   return false;
}
```

and for useKeyboardNav
```js
const action = getMappedAction(event)
if (!action.tabSwitch){
   return;   
}

event.preventDefault();
const doc = ctx.getDocuments()[action.tabSwitch.index + 1];
```

Reply:
refactored